Planit/View/CreatePlanView.swift

```diff
@@ -0,0 +1,86 @@
+//
+//  CreatePlanView.swift
+//  Planit
+//
+//  Created by Shivangi on 06/12/22.
+//
+
+import SwiftUI
+
+struct CreatePlanView: View {
+    
+    let user: AppUser
+    
+    @State var title: String = ""
+    @State var PlanDescription: String = ""
+    @State var PlanType: String = ""
+    @State var completed: Bool = false
+    
+    @ObservedObject var viewModel: planViewModel
+   
+    var body: some View {
+        VStack{
+            
+            PlanButtonStack(viewModel: viewModel)
+                .padding()
+                .padding(.top)
+            
+            PlanTextField(text: $title, placeholder: "Planit")
+                .padding([.horizontal, .bottom])
+            
+            CustomTextField(text: $PlanDescription, placeholder: Text("Description"),    imageName: "line.3.horizontal",
+                            foregroundColor: .gray)
+                            .padding([.horizontal, .bottom])
+            HStack{
+                Button{
+                    viewModel.uploadplan(Plan: plan(ownerUid: user.id ?? "", title: title, description: PlanDescription, planType: viewModel.filterplanSelected == .all ? "Extra" :
+                                                        viewModel.filterplanSelected.rawValue, completed: false))
+                    viewModel.showCreateplanView = false
+                    
+                }label: {
+                    CreatePlanButton()
+                }
+                Button{
+                    viewModel.showCreateplanView = false
+                }label: {
+                    CancelPlanButton()
+                }
+                
+                
+            }
+            Spacer()
+            
+        }//end of Vstack
+        .frame(width: UIScreen.main.bounds.size.width - 50, height: 280)
+        .background(Color(.systemGray5))
+        .cornerRadius(25)
+    }
+}
+
+struct CreatePlanView_Previews: PreviewProvider {
+    static var previews: some View {
+        CreatePlanView(user: appUser01, viewModel: planViewModel())
+    }
+}
+
+
+struct CreatePlanButton: View {
```

Comment:
I love the extraction of the buttons here to keep your other view cleaner!

However, you don't really need a dedicated `View` for this unless you intend to use it in other places too. It instead can just be a var inside the `CreatePlanView`. So the definition line could be something like `var createPlanView: some View {`